### ISSUE TITLE
linter: false positives for empty brace spaces

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#1699]
The following error is reported when it is not an issue:

 ```
   ⚠ eslint-plugin-unicorn(empty-brace-spaces): No spaces inside empty pair of braces allowed
     ╭─[../../microsoft/vscode/src/vs/code/electron-sandbox/issue/issueReporterService.ts:634:1]
 634 │                     }
 635 │ ╭─▶             }).catch(_ => {
 636 │ │                   // Ignore
 637 │ ╰─▶             });
 638 │             }).catch(_ => {
     ╰────
  help: There should be no spaces or new lines inside a pair of empty braces as it affects the overall readability of the code.
  ```
 

We should:
 - port the rest of the cases from https://github.com/sindresorhus/eslint-plugin-unicorn/blob/4cbcabe1b1cbaac0b03cf35196a141c14bf6c8ec/test/empty-brace-spaces.mjs#L1
 - update our lint rule to allow cases where comments exist inbetween the braces to pass

**Comment by Boshen** [comment]
Yes, we need to fix the `outdent` thingy as stated in https://github.com/oxc-project/oxc/issues/1694

**Comment by Boshen** [comment]
https://github.com/oxc-project/oxc/issues/1694 is done, we need to update the tests :-(
